src/Hook/SessionMigrationHook.php

```
@@ -0,0 +1,294 @@
+    public function beforeWrite(string $sessionId, array $data): array
+    {
+        // afterWriteで使用するためデータを保存
+        $this->pendingWrites[$sessionId] = $data;
+        return $data;
+    }
+
+    public function afterWrite(string $sessionId, bool $success): void
+    {
+        if (!$success) {
+            $this->logger->warning('Primary write failed, skipping migration', [
+                'session_id' => SessionIdMasker::mask($sessionId),
```

PHPMD (2.15.0)

Avoid using static access to class '\Uzulla\EnhancedRedisSessionHandler\Support\SessionIdMasker' in method 'afterWrite'. (undefined)

(StaticAccess)